File: outputs/verification/welcome/screenshot2.py
from playwright.sync_api import sync_playwright
import os

HTML_PATH = "/root/projects/brightkidco/outputs/verification/welcome/welcome-01-e4-l2-render.html"
OUT_DIR = "/root/projects/brightkidco/outputs/verification/welcome"

with sync_playwright() as p:
    browser = p.chromium.launch(headless=True)
    
    # 420px full screenshot
    page = browser.new_page(viewport={"width": 420, "height": 800})
    page.goto(f"file://{HTML_PATH}", wait_until="networkidle")
    page.wait_for_timeout(2000)
    path2 = os.path.join(OUT_DIR, "welcome-01-e4-l2-420-full.png")
    page.screenshot(path=path2, full_page=True)
    sz2 = os.path.getsize(path2)
    print(f"420px full: {sz2} bytes")
    page.close()
    
    browser.close()

print("Done")

File: outputs/verification/pp-level-detection/screenshot2.js
const puppeteer = require('puppeteer');
const path = require('path');
const delay = ms => new Promise(r => setTimeout(r, ms));

const HTML_FILE = 'file://' + path.resolve('/root/projects/brightkidco/outputs/verification/pp-level-detection/pp-leveldetect-01-e1.html');
const OUTPUT_DIR = '/root/projects/brightkidco/outputs/verification/pp-level-detection';

(async () => {
  const browser = await puppeteer.launch({
    headless: 'new',
    args: ['--no-sandbox', '--disable-setuid-sandbox', '--disable-dev-shm-usage'],
  });

  // Above-fold screenshot
  const page = await browser.newPage();
  await page.setViewport({ width: 420, height: 568, deviceScaleFactor: 2 });
  await page.goto(HTML_FILE, { waitUntil: 'networkidle0', timeout: 30000 });
  await delay(2000);
  await page.screenshot({
    path: path.join(OUTPUT_DIR, 'pp-leveldetect-01-e1-420-above-fold.png'),
    fullPage: false,
  });
  console.log('Above-fold screenshot saved');

  // Console errors
  const errors = [];
  page.on('console', msg => { if (msg.type() === 'error') errors.push(msg.text()); });
  page.on('pageerror', err => errors.push(err.message));

  // Get metrics
  const metrics = await page.evaluate(() => {
    const root = document.getElementById('root');
    const shell = root?.firstChild;
    return {
      emailWidth: shell ? shell.offsetWidth : 0,
      emailHeight: shell ? shell.offsetHeight : 0,
      consoleErrors: [],
    };
  });

  console.log('Metrics:', JSON.stringify(metrics));
  console.log('Console errors:', JSON.stringify(errors));

  await browser.close();
  console.log('Done');
})();
